src/providers/authenticationProvider.ts

```diff
@@ -215,6 +212,111 @@ export class TerraformCloudAuthenticationProvider implements vscode.Authenticati
+    const choice = await vscode.window.showQuickPick(
+      [
+        {
+          label: 'Stored User Token',
+          detail: 'Use a token stored in the Terraform CLI configuration file',
+        },
+        {
+          label: 'Existing User Token',
+          detail: 'Enter a token manually',
+        },
+        {
+          label: 'Open to generate a User token',
+          detail: 'Open the Terraform Cloud website to generate a new token',
+        },
+      ],
+      {
+        canPickMany: false,
+        ignoreFocusOut: true,
+        placeHolder: 'Choose a method to enter a Terraform Cloud User Token',
+        title: 'HashiCorp Terraform Cloud Authentication',
+      },
+    );
+    if (choice === undefined) {
+      return undefined;
+    }
+
+    // TODO: Change to production URL
+    const terraformCloudURL = 'https://app.staging.terraform.io/app/settings/tokens?source=terraform-login';
+    let token: string | undefined;
+    switch (choice.label) {
+      case 'Open to generate a User token':
+        await vscode.env.openExternal(vscode.Uri.parse(terraformCloudURL));
+        // Prompt for the UAT.
+        token = await vscode.window.showInputBox({
+          ignoreFocusOut: true,
+          placeHolder: 'User access token',
+          prompt: 'Enter an HashiCorp Terraform User Access Token (UAT).',
+          password: true,
+        });
+        break;
+      case 'Existing User Token':
+        // Prompt for the UAT.
+        token = await vscode.window.showInputBox({
+          ignoreFocusOut: true,
+          placeHolder: 'User access token',
+          prompt: 'Enter an HashiCorp Terraform User Access Token (UAT).',
+          password: true,
+        });
+        break;
+      case 'Stored User Token':
+        token = await this.getTerraformCLIToken();
+        break;
+      default:
+        break;
+    }
+
+    return token;
+  }
+
+  private async getTerraformCLIToken() {
+    // detect if stored auth token is present
+    // ~/.terraform.d/credentials.tfrc.json
+    const credFilePath = path.join(os.homedir(), '.terraform.d', 'credentials.tfrc.json');
+    if ((await this.pathExists(credFilePath)) === false) {
+      vscode.window.showErrorMessage(
+        'Terraform credential file not found. Please login using the Terraform CLI and try again.',
+      );
+      return undefined;
+    }
+
+    // read and marshall json file
+    let text: string;
+    try {
+      const content = await vscode.workspace.fs.readFile(vscode.Uri.file(credFilePath));
+      text = Buffer.from(content).toString('utf8');
+    } catch (error) {
+      vscode.window.showErrorMessage(
+        'Failed to read configuration file. Please login using the Terraform CLI and try again',
+      );
+      return undefined;
+    }
+
+    // find app.terraform.io token
+    try {
+      const data = JSON.parse(text);
+      const cred = data.credentials['app.staging.terraform.io'];
```

Comment:
```suggestion
      // TODO: Change to production URL
      const cred = data.credentials['app.staging.terraform.io'];
```

Or do we want to store the hostname (`app.staging.terraform.io`) in a constant at the top of the file and use it here and in line 243?